### ISSUE TITLE
swagger path incorrect when databaseTableName differs from table.name

### ISSUE THREAD

**Issue by joshbooker** [Azure/azure-mobile-apps-node#413]
Repro steps:

Define table in Customer.js like so:
```javascript
  var azureMobileApps = require('azure-mobile-apps');

  // Create a new table definition
  var table = azureMobileApps.table();

  table.dynamicSchema = false;
  //table.name = 'Customers';
  table.schema = 'dbo';

  table.databaseTableName = 'Customers';

  module.exports = table;
```
Now you have a working\endpoint:
/tables/Customer

Problem: swagger uses table.databaseTableName so it has non-working path:
/tables/Customers

**Comment by danderson00** [comment]
Interesting, the problem is a bit deeper down in https://github.com/Azure/azure-mobile-apps-node/blob/master/src/express/tables/index.js#L70. Not sure why databaseTableName takes precedence, need to make sure there are no wider effects of changing it.

Thanks for the report! Should be able to sneak this into a release for tomorrow.

**Comment by danderson00** [comment]
There is definitely a bigger can of worms here. SQL statements are generated using table.name (not a big change), but the complicating factor is that so does azure-query-js (i.e. read queries). More thought required.

**Comment by joshbooker** [comment]
In layman's terms, correct me if wrong...databaseTableName takes precedence at the deeper level for good reason so the schema queries, etc. work on db.  

Then where is the table route defined?  I assume that doesn't use the same precedent and instead picks up table.name  || .js file name. Can the swagger gen use the route name?

**Comment by danderson00** [comment]
Yep, precisely, and a very good question. The router is mounted when the table is added, from the name that is passed to the add function - https://github.com/Azure/azure-mobile-apps-node/blob/master/src/express/tables/index.js#L41. This is either what is explicitly passed in user code or the file name passed by import. This means that if a table name is explicitly set in the table definition, the route is wrong. Table name was always a bit of an issue - the name parameter on the add function is mandatory (initial assumption), but it's still possible to override in a table definition (added later). 

We need to fix this properly, i.e. make table.name and table.databaseTableName distinct, and formalise how the table name is set. Watch this space.

**Comment by joshbooker** [comment]
Thanks. I wonder if there's a temp quick fix for swagger. It looks like in the preceding line 40 that the name may persist in configuration.tables. 

configuration.tables[name] = definition;
If I understand the code, it seems this would result in object

configuration.tables.customer having property .name= 'Customers'

Is that the same tables schema object used to create swagger paths?  If so can you get 'customer' instead of 'customers'?

**Comment by danderson00** [comment]
Will publish a new version this afternoon. The referenced PR should resolve the issue, swagger should now emit the correct paths.

**Comment by joshbooker** [comment]
Thanks Dale.  Now I get error: creating table 'Customer' there is already a table in database 'Customers'
'''
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\josh>cd c:\source\nwmobi

c:\Source\nwmobi>npm uninstall azure-mobile-apps
unbuild azure-mobile-apps@2.1.0

c:\Source\nwmobi>npm install Azure/azure-mobile-apps-node
azure-mobile-apps@2.1.1 node_modules\azure-mobile-apps
├── azure-odata-sql@0.0.3
├── es6-promise@3.1.2
├── node-uuid@1.4.7
├── azure-query-js@0.0.3 (esprima@2.7.2)
├── winston@2.2.0 (cycle@1.0.3, stack-trace@0.0.9, isstream@0.1.2, eyes@0.1.8, p
kginfo@0.3.1, async@1.0.0, colors@1.0.3)
├── body-parser@1.15.0 (bytes@2.2.0, content-type@1.0.1, depd@1.1.0, on-finished
@2.3.0, http-errors@1.4.0, raw-body@2.1.6, debug@2.2.0, qs@6.1.0, type-is@1.6.12
, iconv-lite@0.4.13)
├── jsonwebtoken@5.7.0 (xtend@4.0.1, ms@0.7.1, jws@3.1.3)
├── azure-sb@0.10.2 (mpns@2.0.1, underscore@1.4.4, wns@0.5.3, azure-common@0.9.1
6)
└── mssql@3.2.1 (generic-pool@2.4.2, promise@7.1.1, tedious@1.14.0)

c:\Source\nwmobi>node --debug app.js
debugger listening on port 5858
2016-04-22T22:08:17.925Z - verbose: Adding table definition for Customer
2016-04-22T22:08:17.940Z - verbose: Adding table definition for Orders
2016-04-22T22:08:17.945Z - verbose: Adding table definition for OrderDetail
2016-04-22T22:08:17.948Z - verbose: Adding table definition for Product
2016-04-22T22:08:17.951Z - verbose: Adding table definition for TodoItem
2016-04-22T22:08:18.501Z - info: Creating table Customer
2016-04-22T22:08:19.173Z - error: Error occurred creating table Customer: Reques
tError: There is already an object named 'Customers' in the database.
    at c:\Source\nwmobi\node_modules\azure-mobile-apps\node_modules\mssql\lib\te
dious.js:724:17
    at Connection.emit (events.js:95:17)
    at Parser.<anonymous> (c:\Source\nwmobi\node_modules\azure-mobile-apps\node_
modules\mssql\node_modules\tedious\lib\connection.js:204:16)
    at Parser.emit (events.js:95:17)
    at Parser.<anonymous> (c:\Source\nwmobi\node_modules\azure-mobile-apps\node_
modules\mssql\node_modules\tedious\lib\token\token-stream-parser.js:42:15)
    at Parser.emit (events.js:95:17)
    at readableAddChunk (c:\Source\nwmobi\node_modules\azure-mobile-apps\node_mo
dules\mssql\node_modules\tedious\node_modules\readable-stream\lib\_stream_readab
le.js:198:18)
    at Parser.Readable.push (c:\Source\nwmobi\node_modules\azure-mobile-apps\nod
e_modules\mssql\node_modules\tedious\node_modules\readable-stream\lib\_stream_re
adable.js:157:10)
    at Parser.Transform.push (c:\Source\nwmobi\node_modules\azure-mobile-apps\no
de_modules\mssql\node_modules\tedious\node_modules\readable-stream\lib\_stream_t
ransform.js:123:32)
    at doneParsing (c:\Source\nwmobi\node_modules\azure-mobile-apps\node_modules
\mssql\node_modules\tedious\lib\token\stream-parser.js:87:17)
2016-04-22T22:08:19.207Z - error: Error occurred during table initialization Req
uestError: There is already an object named 'Customers' in the database.
    at c:\Source\nwmobi\node_modules\azure-mobile-apps\node_modules\mssql\lib\te
dious.js:724:17
    at Connection.emit (events.js:95:17)
    at Parser.<anonymous> (c:\Source\nwmobi\node_modules\azure-mobile-apps\node_
modules\mssql\node_modules\tedious\lib\connection.js:204:16)
    at Parser.emit (events.js:95:17)
    at Parser.<anonymous> (c:\Source\nwmobi\node_modules\azure-mobile-apps\node_
modules\mssql\node_modules\tedious\lib\token\token-stream-parser.js:42:15)
    at Parser.emit (events.js:95:17)
    at readableAddChunk (c:\Source\nwmobi\node_modules\azure-mobile-apps\node_mo
dules\mssql\node_modules\tedious\node_modules\readable-stream\lib\_stream_readab
le.js:198:18)
    at Parser.Readable.push (c:\Source\nwmobi\node_modules\azure-mobile-apps\nod
e_modules\mssql\node_modules\tedious\node_modules\readable-stream\lib\_stream_re
adable.js:157:10)
    at Parser.Transform.push (c:\Source\nwmobi\node_modules\azure-mobile-apps\no
de_modules\mssql\node_modules\tedious\node_modules\readable-stream\lib\_stream_t
ransform.js:123:32)
    at doneParsing (c:\Source\nwmobi\node_modules\azure-mobile-apps\node_modules
\mssql\node_modules\tedious\lib\token\stream-parser.js:87:17)

c:\Source\nwmobi>'
''

**Comment by danderson00** [comment]
Sorry about that, I missed a test case. I just published 2.1.2 that should fix the problem.
